Source/WebKit/Platform/IPC/Connection.h

```diff
@@ -639,7 +642,6 @@ template<typename T> Connection::SendSyncResult<T> Connection::sendSync(T&& mess
 
 template<typename T> bool Connection::waitForAndDispatchImmediately(uint64_t destinationID, Timeout timeout, OptionSet<WaitForOption> waitForOptions)
 {
-    RELEASE_ASSERT(RunLoop::isMain());
```

Comment:
Any reason not to assert that we're on the connections run loop for these?

Reply:
The assertion must make a check "do we know what the run loop of this connection in the first place", e.g. isValid().
So it's done in [1] above.